core/Access.php

```diff
@@ -90,34 +85,53 @@ public static function getInstance()
     }
 
     /**
-     * Returns the list of the existing Access level.
-     * Useful when a given API method requests a given acccess Level.
-     * We first check that the required access level exists.
-     *
-     * @return array
+     * @var CapabilitiesProvider
      */
-    public static function getListAccess()
-    {
-        return self::$availableAccess;
-    }
+    protected $capabilityProvider;
+
+    /**
+     * @var RolesProvider
+     */
+    private $roleProvider;
 
     /**
      * Constructor
      */
-    public function __construct()
+    public function __construct(RolesProvider $roleProvider = null, CapabilitiesProvider $capabilityProvider = null)
     {
+        if (!isset($roleProvider)) {
```

Comment:
was needed as some system tests otherwise fail... it would try to use DI before the environment is set up etc.